### ISSUE TITLE
Error middleware changes status to 200 if error page not tiny

### ISSUE THREAD

**Issue by MetaSteel** [traefik/traefik#8926]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I embedded a base64 encoded image into my 404 error page. This increased the static file size from ~1k to ~100k, after the addition, the server began returning status:200 instead of the expected status:404. removing the image data, returned the expected status:404.

To investigate further I left the data in the file, but commented out,. The server began returning status:200 again. I believe the file size is the only factor.

### What did you see instead?

seeing 200, not 404 when error page is larger then some threshold.

### What version of Traefik are you using?

traefik:v2.6

### What is your environment & configuration?

```yaml
version: '3.7'
services:
  traefik:
    image: traefik:v2.6
    command:
      # Docker swarm configuration
      - --providers.docker.endpoint=http://docker-socket:2375
      - --providers.docker.swarmMode=true
      - --providers.docker.exposedbydefault=false
      - --providers.docker.network=traefik

      # Global HTTP -> HTTPS
      - --entrypoints.web.address=:80
      - --entrypoints.web.http.redirections.entryPoint.to=websecure
      - --entrypoints.web.http.redirections.entryPoint.scheme=https
      - --entrypoints.web.http.redirections.entryPoint.permanent=true
      
      # Global TLS
      - --entrypoints.websecure.address=:443
      - --entrypoints.websecure.http.middlewares=globalerror

    ports:
      - target: 80 # web
        published: 80
        mode: host
      - target: 443 # tls
        published: 443
        mode: host
    volumes:
     - /mnt/tank/persist/site/traefik/production/config/:/letsencrypt
    networks:
     - docker-socket
     - traefik
    logging:
      driver: json-file
    deploy:
      labels:
        traefik.enable: 'true'
        
        # Create error middleware
        traefik.http.middlewares.globalerror.errors.status: '404'
        traefik.http.middlewares.globalerror.errors.service: caddy-server
        traefik.http.middlewares.globalerror.errors.query: '/404.htm'

      restart_policy:
        condition: on-failure

networks:
  docker-socket:
    external: true
  traefik:
    external: true

```
Heavily redacted, but specific docker stack file Illustrating how I'm using Traefik.
The static file host is Caddy.

**Comment by kevinpollet** [comment]
Hello @MetaSteel,

Thanks for your interest in Traefik,

We tried to reproduce the issue but without any success,

please could you provide a reproducible use case?

**Comment by mpl** [comment]
@MetaSteel 
You can still send a self-contained repro if you have narrowed the problem even more, but we actually eventually reproduced, so we do have something to debug with now.

**Comment by MetaSteel** [comment]
@mpl
I'm glad you got it to reproduce. I haven't done anymore narrowing down and I didn't have time yet to try to make a stand alone. If I can find time later I'll get another example up.

**Comment by mpl** [comment]
we found the problem, and several variants for the fix. We just need to think a bit harder about which variant is the best one.

**Comment by traefiker** [comment]
Closed by #8932.
